bin/reth/src/commands/node/mod.rs

```diff
@@ -181,9 +180,7 @@ impl<Ext: clap::Args + fmt::Debug> NodeCommand<Ext> {
             node_config = node_config.with_unused_ports();
         }
 
-        let builder = NodeBuilder::new(node_config)
-            .with_database(database)
-            .with_launch_context(ctx.task_executor);
+        let builder = NodeBuilder::new(node_config).with_database(database).with_launch_context();
```

Comment:
Is with launch context needed now?

Reply:
Don't think so. Working on a commit now that removes a bunch of code like this. Will see how it goes

Reply:
While I can remove the task executor from the launch context, the launch context is still required for other functionality such as `launch_node()`. E.G. if I use builder directly:
```rust
  --> bin/reth/src/main.rs:24:30
   |
24 |         let handle = builder.launch_node(EthereumNode::default()).await?;
   |                              ^^^^^^^^^^^ method not found in `NodeBuilder<Arc<DatabaseEnv>>`
```

It might be possible to move the `WithLaunchContext` impl into the builder. No idea if this is desirable etc, I'm not familiar.

Would suggest to keep `launch_with_context()` here for now and if we need a followup issue/pr, can do. This current PR might be large enough if we decide to keep the approach shown so far.